train.py

```diff
@@ -52,41 +52,47 @@
-                visualizer.display_current_results(model.get_current_visuals(), epoch, save_result,params=model.get_display_param())
-
-            if total_iters % opt.print_freq == 0:    # print training losses and save logging information to the disk
-                losses = model.get_current_losses()
-                t_comp = (time.time() - iter_start_time) / opt.batch_size
-                visualizer.print_current_losses(epoch, epoch_iter, losses, t_comp, t_data)
-                if opt.display_id > 0:
-                    visualizer.plot_current_losses(epoch, float(epoch_iter) / dataset_size, losses)
-
-            if total_iters % opt.save_latest_freq == 0:   # cache our latest model every <save_latest_freq> iterations
-                print('saving the latest model (epoch %d, total_iters %d)' % (epoch, total_iters))
-                save_suffix = 'iter_%d' % total_iters if opt.save_by_iter else 'latest'
-                model.save_networks(save_suffix)
-
-            if total_iters % opt.fid_every==0 and opt.compute_fid:
-                model.compute_fid(epoch,total_iters)
-                if opt.display_id > 0:
-                    fids=model.get_current_fids()
-                    visualizer.plot_current_fid(epoch, float(epoch_iter) / dataset_size, fids)
-
+            if display_all :
+                if total_iters % opt.display_freq == 0:   # display images on visdom and save images to a HTML file
+                    save_result = total_iters % opt.update_html_freq == 0
+                    model.compute_visuals()
+                    visualizer.display_current_results(model.get_current_visuals(), epoch, save_result,params=model.get_display_param())
+
+                if total_iters % opt.print_freq == 0:    # print training losses and save logging information to the disk
+                    losses = model.get_current_losses()
+                    t_comp = (time.time() - iter_start_time) / opt.batch_size
+                    visualizer.print_current_losses(epoch, epoch_iter, losses, t_comp, t_data)
+                    if opt.display_id > 0:
+                        visualizer.plot_current_losses(epoch, float(epoch_iter) / dataset_size, losses)
+
+                if total_iters % opt.save_latest_freq == 0:   # cache our latest model every <save_latest_freq> iterations
+                    print('saving the latest model (epoch %d, total_iters %d)' % (epoch, total_iters))
+                    save_suffix = 'iter_%d' % total_iters if opt.save_by_iter else 'latest'
+                    model.save_networks(save_suffix)
+
+                if total_iters % opt.fid_every==0 and opt.compute_fid:
+                    model.compute_fid(epoch,total_iters)
+                    if opt.display_id > 0:
+                        fids=model.get_current_fids()
+                        visualizer.plot_current_fid(epoch, float(epoch_iter) / dataset_size, fids)
+
+            if model.niter % opt.iter_size == 0 :
```

Comment:
Do this check at the beggining of the loop so you don't need `display_all`

Reply:
Done :)